### PR TITLE
fixes #274 - add Range to data request objects

### LINKED ISSUE #274: No way to specify a range on client data requests
The request objects should allow for a range to be specified.

### PR DESCRIPTION
**Pull request by dcrissman**

### CODE REVIEW

File: core/src/main/java/com/redhat/lightblue/client/request/data/DataFindRequest.java [hunk 1]
```diff
@@ -35,11 +34,23 @@ public void where(Query queryExpression) {
...
         projection = Projection.project(projections);
+        this.begin = begin;
+        this.end = end;
     }
 
     public void select(Projection... projection) {
+        select(projection, null, null);
+    }
+
+    public void select(Projection[] projection, Integer begin, Integer end) {
```
Comment (dcrissman) [inline core/src/main/java/com/redhat/lightblue/client/request/data/DataFindRequest.java]: This does sour syntactical sugar somewhat. For this variation you would need to declare the array. I just don't like how (int, int, Projection...) read, but I am open to opposition on this point. This is backwards compatible as the method variation is net new.

Example:
```java
select(new Projection[]{Projection1, Projection2}, int, int);
```

**Comment by dcrissman** [comment]
Depends on https://github.com/lightblue-platform/lightblue-core/issues/614

**Comment by dcrissman** [comment]
Pending https://github.com/lightblue-platform/lightblue-mongo/pull/212

**Comment by dcrissman** [comment]
Ready for code review

**Comment by bserdar** [comment]
rebase pls

**Comment by dcrissman** [comment]
Changes from master have been merged in. This is ready for review.

**Pull request merged**